tasty-plutus/src/Test/Tasty/Plutus/Internal/TestScript.hs

```diff
@@ -0,0 +1,288 @@
+getWrappedMintingPolicy ::
+  WrappedMintingPolicy -> (BuiltinData -> BuiltinData -> ())
+getWrappedMintingPolicy (WrappedMintingPolicy mp) = mp
+
+{- | 'mkTestValidator' provide the way for creating 'TestScript' with
+ the given 'Validator', parameterized with proper datum and redeemer types.
+
+  = Usage
+
+ > testValidator :: TestScript ( 'ForSpending SomeType SomeOtherType)
+ > testValidator =
+ >  mkTestValidator
+ >    $$(compile [||myValidator||])
+ >    $$(compile [||toTestValidator||])
+
+ @since 6.0
+-}
+mkTestValidator ::
+  forall (datum :: Type) (redeemer :: Type) (ctx :: Type).
+  CompiledCode (datum -> redeemer -> ctx -> Bool) ->
+  CompiledCode ((datum -> redeemer -> ctx -> Bool) -> WrappedValidator) ->
+  TestScript ( 'ForSpending datum redeemer)
+mkTestValidator v wr =
+  TestValidator $
+    mkValidatorScript $
+      applyCode $$(compile [||getWrappedValidator||]) $
+        wr `applyCode` v
+
+{- | 'mkTestValidatorUnsafe' provide the way for creating 'TestScript' with
+ the given 'Validator', parameterized with any datum and redeemer types.
+ You can use it in case you going to test something tricky and need to break
+ types compatibility.
+
+  = Usage
+
+ > myValidator :: SomeDatumType -> SomeRedeemerType -> ScriptContext -> Bool
+ > myValidator d r ctx = ...
+
+ > testValidator ::
+ >     TestScript ( 'ForSpending SomeOtherDatumType SomeOtherRedeemerType)
+ > testValidator =
+ >  mkTestValidatorUnsafe
+ >    $$(compile [||myValidator||])
+ >    $$(compile [||toTestValidator||])
+
+ @since 6.0
+-}
+mkTestValidatorUnsafe ::
+  forall (d :: Type) (r :: Type) (datum :: Type) (redeemer :: Type) (ctx :: Type).
+  CompiledCode (datum -> redeemer -> ctx -> Bool) ->
+  CompiledCode ((datum -> redeemer -> ctx -> Bool) -> WrappedValidator) ->
+  TestScript ( 'ForSpending d r)
+mkTestValidatorUnsafe v wr =
+  TestValidator $
+    mkValidatorScript $
+      applyCode $$(compile [||getWrappedValidator||]) $
+        wr `applyCode` v
+
+{- | 'mkTestMintingPolicy' provide the way for creating 'TestScript' with
+ the given 'MintingPolicy', parameterized with proper redeemer type.
+
+  = Usage
+
+ > testMintingPolicy :: TestScript ( 'ForMinting SomeType)
+ > testMintingPolicy =
+ >  mkTestMintingPolicy
+ >    $$(compile [||myMintingPolicy||])
+ >    $$(compile [||toTestMintingPolicy||])
+
+ @since 6.0
+-}
+mkTestMintingPolicy ::
+  forall (redeemer :: Type) (ctx :: Type).
+  CompiledCode (redeemer -> ctx -> Bool) ->
+  CompiledCode ((redeemer -> ctx -> Bool) -> WrappedMintingPolicy) ->
+  TestScript ( 'ForMinting redeemer)
+mkTestMintingPolicy mp wr =
+  TestMintingPolicy $
+    mkMintingPolicyScript $
+      applyCode $$(compile [||getWrappedMintingPolicy||]) $
+        wr `applyCode` mp
+
+{- | 'mkTestMintingPolicyUnsafe' provide the way for creating 'TestScript' with
+ the given 'MintingPolicy', parameterized with any redeemer type.
+ You can use it in case you going to test something tricky
+ and need to break types compatibility.
```

Comment:
Similar phrasing changes (and warning) as for 'mkTestValidatorUnsafe' above would be good.